core/meta/inc/TClass.h

```diff
@@ -132,6 +132,26 @@ friend class TStreamerInfo;
                         // in ROOT Meta w/o interpreter information
    };
 
+   // "Typed" pointer that recalls how TClass::New allocated the object.
+   // It is returned by TClass:NewObject and should be passed to TClass::DeleteArray or TClass::Destructor
+   // to delete the object.
+   // It is also used in TVirtualCollectionProxy for the same reasons.
+   class ObjectPtr
+   {
+      void *fPtr = nullptr;
+
+      TVirtualStreamerInfo *fAllocator = nullptr;
+
+   public:
+      ObjectPtr(void *ptr = nullptr, TVirtualStreamerInfo *allocator = nullptr) : fPtr(ptr), fAllocator(allocator) {}
```

Comment:
Isn't it "either no args or both"? If so please do;
```suggestion
      ObjectPtr() = default;
      ObjectPtr(void *ptr, TVirtualStreamerInfo *allocator) : fPtr(ptr), fAllocator(allocator) {} 
```

Reply:
> Isn't it "either no args or both"? If so please do;

In the current code (possibly to pedantic) it is either one or two.  The case one is use for implicit creation from a void *, for example:
```
ObjectPtr NewObject(....) {
....
    return fNew(....);
....
}
```